pkg/operation/botanist/resources.go

```diff
@@ -45,6 +45,8 @@ func (b *Botanist) DeployReferencedResources(ctx context.Context) error {
 			return fmt.Errorf("object not found %v", resource.ResourceRef)
 		}
 
+		obj = unstructuredutils.FilterMetadata(obj, "finalizers", "labels", "annotations")
```

Comment:
Q: can it be the case that the extension relies on a label/annotation on the referenced resource. I don't have in mind existing case but it sounds reasonable to keep the original labels and annotations. WDYT?